rbx_types/src/attributes/writer.rs

```diff
@@ -0,0 +1,136 @@
+use super::*;
+
+use crate::{basic_types::*, variant::Variant};
+
+use std::{borrow::Borrow, io::Write};
+
+fn write_f32<W: Write>(mut writer: W, n: f32) {
+    writer
+        .write_all(&n.to_le_bytes().to_vec())
```

Comment:
Definitely do not allocate a `Vec` for every four bytes we need to write, that's very excessive!

Use `&n.to_le_bytes()[..]` instead.